File: ingester/internal/events/burn_event.go
package events

type BurnEvent struct {
	BaseEvent
	Sender    string `json:"sender"`
	Recipient string `json:"recipient"`
	Amount0   string `json:"amount0"` // Token0 amount removed from pool
	Amount1   string `json:"amount1"` // Token1 amount removed from pool
}

func (e BurnEvent) ToMap() map[string]interface{} {
	m := e.BaseEvent.ToMap()
	m["sender"] = e.Sender
	m["recipient"] = e.Recipient
	m["amount0"] = e.Amount0
	m["amount1"] = e.Amount1
	return m
}

File: aggregator/src/main/java/com/web3analytics/models/BurnEvent.java
package com.web3analytics.models;

/**
 * Burn event from dex-liquidity-events topic.
 * Represents liquidity removal (token amounts removed).
 */
public record BurnEvent(
        String eventId,
        long blockNumber,
        long blockTimestamp,
        String transactionHash,
        int logIndex,
        String pairAddress,
        String token0,
        String token1,
        String token0Symbol,
        String token1Symbol,
        String sender,
        String recipient,
        String amount0,
        String amount1,
        long eventTimestamp
) implements DexEvent {}
